credentials/alts/alts_test.go

```diff
@@ -287,6 +296,50 @@ func (s) TestCheckRPCVersions(t *testing.T) {
 	}
 }
 
+func TestFullHandshake(t *testing.T) {
+	// If GOMAXPROCS is set to less than 2, do not run this test. This test
+	// requires at least 2 goroutines to succeed (one goroutine where a
+	// server listens, another goroutine where a client runs).
+	if runtime.GOMAXPROCS(0) < 2 {
+		return
+	}
+
+	// Start the fake handshaker service and the server.
+	var wait sync.WaitGroup
+	defer wait.Wait()
+	stopHandshaker, handshakerAddress := startFakeHandshakerService(t, &wait)
+	defer stopHandshaker()
+	stopServer, serverAddress := startServer(t, handshakerAddress, &wait)
+	defer stopServer()
+
+	// Connect to the server, authenticating with ALTS and the fake handshaker service.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	vmOnGCP = true
```

Comment:
I don't see where this is used. If this is a package global, then please set it at the top of the test, and reset it to its original value with a `defer`, and have a small comment about why it is being set etc.

Reply:
This is a global variable defined in [`alts.go`](https://github.com/grpc/grpc-go/blob/3489bb7d51741618f1b1e0c723ca6dc905118e3b/credentials/alts/alts.go#L58). I added a comment in the test explaining why we have to reset it to true.